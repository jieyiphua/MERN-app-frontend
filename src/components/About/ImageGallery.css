.gallery-outer-container {
  width: 100%;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-start;
}

.image-thumbnail {
  width: 250px;
  height: 250px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin: 15px 15px;
}

.images {
  max-width: 250px;
  max-height: 250px;

}