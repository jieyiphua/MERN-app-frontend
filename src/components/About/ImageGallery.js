import React from "react";
import "./ImageGallery.css";
import axios from "axios";

class ImageGallery extends React.Component {
  state = {
    images: null
  };

  async componentDidMount() {
    const response = await axios.get(
      "https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/images/get-objects"
    );
    console.log(response);
    this.setState({
      images: response.data
    });

    
  }

  

  render() {
    const imageState = this.state.images
    if (!imageState) {
      return null
    } else {
      const imageArr = imageState.imageUrls
      console.log(imageArr)
      return (
        
        <div className="main-container">
          <div className="inner-main-container">
          <h1 className="bcmaPageHeaderH1">Image Gallery</h1>
            <div className="gallery-outer-container">
            
            {imageArr.slice(1).map((url ) => (
              <div className="image-thumbnail">   
                <img className="images" src={url}></img>
              </div>
          ))}



      


            </div>
          </div>
        </div>
        
      );
    }
  }
}

export default ImageGallery;
