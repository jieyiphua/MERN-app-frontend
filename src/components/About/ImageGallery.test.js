import ImageGallery from "./ImageGallery";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<ImageGallery />", () => {
  it("renders component", () => {
    const wrapper = shallow(<ImageGallery />);
    expect(wrapper).toMatchSnapshot();
  });
});
