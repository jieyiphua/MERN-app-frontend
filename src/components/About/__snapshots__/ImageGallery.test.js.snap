// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageGallery /> renders component 1`] = `""`;
